From fc02f150177f9e3b307eb31a8e7e296809c22636 Mon Sep 17 00:00:00 2001
From: Hendrik Leppkes <h.leppkes@gmail.com>
Date: Fri, 17 Aug 2012 20:45:25 +0200
Subject: [PATCH 11/62] h264: flag interlaced-coded frames as such when
 sei_pic_struct indicates otherwise

According to the H.264 standard, Annex D, Table D-1, field_pic_flag shall be 0 when sei_pic_struct signals progressive. In the case when this restriction is not fullfilled, its safer to assume the field is actually interlaced.
Not doing so can cause a renderer to assume the frame is progressive and not deinterlace it, even if required.
---
 libavcodec/h264.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libavcodec/h264.c b/libavcodec/h264.c
index f8f99e2..275d796 100644
--- a/libavcodec/h264.c
+++ b/libavcodec/h264.c
@@ -773,6 +773,7 @@ static void decode_postinit(H264Context *h, int setup_finished)
     if (h->sps.pic_struct_present_flag) {
         switch (h->sei_pic_struct) {
         case SEI_PIC_STRUCT_FRAME:
+            cur->f.interlaced_frame = FIELD_OR_MBAFF_PICTURE(h);
             break;
         case SEI_PIC_STRUCT_TOP_FIELD:
         case SEI_PIC_STRUCT_BOTTOM_FIELD:
-- 
2.3.0

